src/vos/vos_aggregate.c

```diff
@@ -1685,6 +1797,22 @@ join_merge_window(daos_handle_t ih, struct agg_merge_window *mw,
 		D_ASSERT(merge_window_status(mw) == MW_FLUSHED);
 	}
 
+	if (entry->ie_vis_flags & VOS_VIS_FLAG_REMOVE) {
```

Comment:
Simply check "if (remove)"?

Reply:
yeah, I added the flag after this code was added.  It's more obvious to a reviewer :)